pkgs/build-support/rust/default.nix

```diff
@@ -11,7 +11,13 @@
 }:
 
 { name ? "${args.pname}-${args.version}"
-, cargoSha256 ? "unset"
+
+  # SRI hash
+, cargoHash ? ""
+
+  # Legacy hash
+, cargoSha256 ? ""
```

Comment:
Is this just a simple migration to a new var name / style without any real functional difference?  Would be worth updating the section of the manual with this.

Reply:
I know that SRI hashes are already used in `nixpkgs` but I am not sure if we endorse them yet. I am also not sure whether with stable Nix will actually generate them for you.

Reply:
Seems to work:

```
❯ nix --version
nix (Nix) 2.3.8
❯ nix-build -A broot
these derivations will be built:
  /nix/store/w86388xb4h1250wzyyzli61ji7gh6y8x-broot-1.0.4-vendor.tar.gz.drv
  /nix/store/zswggbvzvi858czcqr7cvblzk84zk58k-broot-1.0.4.drv
building '/nix/store/w86388xb4h1250wzyyzli61ji7gh6y8x-broot-1.0.4-vendor.tar.gz.drv'...
error: --- Error --- nix-daemon
hash mismatch in fixed-output derivation '/nix/store/w86388xb4h1250wzyyzli61ji7gh6y8x-broot-1.0.4-vendor.tar.gz.drv':
  specified: sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
     got:    sha256-ikx2jlSg+kzhZce3C2+7sKgMokJ24dmHpw8d+eSIe7E=
error: error: --- Error --- nix-daemon
1 dependencies of derivation '/nix/store/zswggbvzvi858czcqr7cvblzk84zk58k-broot-1.0.4.drv' failed to build
```

Reply:
@bhipple any objections against merging this? The `hash` attribute of `fetchurl` is not documented yet either, but this change makes things consistent with `fetchurl`. (It's kinda weird to place SRI hashes in `cargoSha256` as well ;) .)

Reply:
Need documentation update.

Reply:
> I know that SRI hashes are already used in `nixpkgs` but I am not sure if we endorse them yet. I am also not sure whether with stable Nix will actually generate them for you.

It won't generate it but you can convert it and it will understand it.